### PR TITLE
use bash to execute pre or post change commands

### LINKED ISSUE #4: Change Hooks
Pre / Post change arbitrary bash commands

### PR DESCRIPTION
**Pull request by grep-ir**
closes #4

### CODE REVIEW

File: sinkworker/sinkworker.go [hunk 1]
```diff
@@ -181,4 +189,18 @@ func write(cfg sink.SinkConfig, content string) {
 	if err != nil {
 		log.Panic(err)
 	}
+}
+
+func runCommand(command string) error {
+	// Check that bash is available
+	_, err := exec.LookPath("bash")
```
Comment (chrisjohnson) [inline sinkworker/sinkworker.go]: I think using bash to execute here would be a surprising behavior, I would expect it to just run whatever command I specify, and if I wanted to make use of bash, I would put that into my command

File: sinkworker/sinkworker.go [hunk 2]
```diff
@@ -92,17 +93,24 @@ func process(ctx context.Context, cfg sink.SinkConfig) error {
 	// Compare
 	changed := oldContent != newContent
 
-	// Conditionally to pre / post change
-	// Conditionally write
+	// If a change was detected run pre/post commands and write the new file
 	if changed {
 		if cfg.PreChange != "" {
-			log.Println("TODO: prechange hooks")
+			log.Println("Executing PreChange command")
```
Comment (chrisjohnson) [inline sinkworker/sinkworker.go]: I think we should either drop the logs here, or the log in runCommand, right now it's basically printing two lines for each execution

Comment (grep-ir) [reply sinkworker/sinkworker.go]: fixed

File: sinkworker/sinkworker.go (hunk 1 above)
Comment (grep-ir) [reply sinkworker/sinkworker.go]: will use `sh` instead

**Pull request merged**